MinPlatformPkg/Test/Library/TestPointLib/MmTestPointCommunication.c

```diff
@@ -207,45 +207,42 @@ MmiHandlerTestPointCopyData (
 **/
 VOID
 MmTestPointMmiHandlerGetDataByOffset (
-  IN MMI_HANDLER_TEST_POINT_PARAMETER_GET_DATA_BY_OFFSET     *MmiHandlerTestPointParameterGetDataByOffset
+  IN VOID     *MmiHandlerTestPointParameterGetDataByOffset

```

Comment:
Why was the parameter type made `VOID` when its just cast back to `MMI_HANDLER_TEST_POINT_PARAMETER_GET_DATA_BY_OFFSET` in the function?

Reply:
This is what was causing the issue with the comm buffer.  When having the input as a pointer to a `MMI_HANDLER_TEST_POINT_PARAMETER_GET_DATA_BY_OFFSET` struct I was getting a generic fault because the Security cookie was being overwritten.  This is because the security token was being written right after the function's input which had a definite size.  I wanted to write beyond the empty array I assigned to represent the data in the struct which overwrote the cookie.  The `VOID` buffer had the size we passed into the MMI which was the size assigned to fit the data so we don't run into this problem.

Reply:
A pointer type is the same size on the stack regardless of what it points to. A `VOID *` and a `MMI_HANDLER_TEST_POINT_PARAMETER_GET_DATA_BY_OFFSET *` will both consume a size of `UINTN`. So, the parameter type here shouldn't matter. Making it `MMI_HANDLER_TEST_POINT_PARAMETER_GET_DATA_BY_OFFSET *` will avoid an unnecessary assignment in the function that transfers the pointer value.

Looking at your diff, it seems the problem was that you previously had an independent local structure on the stack that overflowed in the `CopyMem()`:

```c
  MMI_HANDLER_TEST_POINT_PARAMETER_GET_DATA_BY_OFFSET    MmiHandlerTestPointGetDataByOffset;

  ...

  CopyMem (
    &MmiHandlerTestPointGetDataByOffset,
    MmiHandlerTestPointParameterGetDataByOffset,
    MmiHandlerTestPointParameterGetDataByOffset->DataSize
    );
```

Reply:
I tested to make sure the change still works and it does so I've pushed it.  Closing this discussion.